### PR TITLE
fix the doc builds, add a release-note

### PR DESCRIPTION
**Pull request by tayritenour**
## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/release-notes/deepspeed-autotune.rst
Comment (garrett361) [inline docs/release-notes/deepspeed-autotune.rst:1]: Is it possible to point the release note to the docs? No problem if not; not sure what's possible here.

**Review by garrett361 — APPROVED** [review]
LGTM, just one q

File: docs/release-notes/deepspeed-autotune.rst
Comment (tayritenour) [reply docs/release-notes/deepspeed-autotune.rst:1]: So I would, but the README for release notes explicitly says that I shouldn't: 
```
[...]
Be sure to highlight API changes and backward incompatibility, including steps needed to upgrade safely.

       Do not include:

       * links to Determined documentation
       * customer identifiers
       * internal project status or plans
       * Jira issue or PR identifiers
```

**Review by tara-det-ai — APPROVED** [review]
LGTM

**Review by tara-det-ai — APPROVED** [review]
LGTM

**Pull request merged**